### ISSUE TITLE
Why is Video reader returning garbage labels all of a sudden?

### ISSUE THREAD

**Issue by RafayAK** [NVIDIA/DALI#4397]
Hi, I have been using `fn.readers.video` just fine but all of a sudden it returns garbage labels that crash the training.

Here's an example (I started printing for debug)
```
logits tensor([[-0.1862, -0.5269, -0.8141, -1.7697, -0.7149, -0.2036],
        [-0.0952, -0.5359, -1.0048, -1.9695, -0.7174, -0.4702],
        [-0.1614, -0.5391, -0.9486, -1.5993, -0.7009, -0.2566],
        [ 1.8298, -2.9449, -3.5812, -5.9424, -3.8149,  2.0153]],
       device='cuda:0', grad_fn=<AddmmBackward0>)

y tensor([0, 4, 4, 0], device='cuda:0')

logits tensor([[-0.2480, -0.2945, -0.3510, -1.4954, -0.4261, -0.7061],
        [-0.9072,  0.0794,  0.4878, -1.7201, -0.6338, -1.2064],
        [-0.4393, -0.0663, -0.2625, -1.7640, -0.3741, -1.2005],
        [-0.4431, -0.0608, -0.1595, -1.7119, -0.2972, -1.2982]],
       device='cuda:0', grad_fn=<AddmmBackward0>)

y tensor([4633364282879029010, 4633373665382943948, 4633383047886858887,
                          4], device='cuda:0')   ==============> HERE IS THE PROBLEM!!!!

Epoch 1/9  ━━━━━━━━━━━━━━━━━╸━━━━━━━━━━━━━━━━━━━━━━ 57/127 0:02:18 • 0:02:53 0.40it/s loss: 1.88 v_num: yap0 val/loss: 2.658 val/acc: 0.316 val/acc_best: 0.316 train/loss: 2.005 train/acc: 0.229 ../aten/src/ATen/native/cuda/Loss.cu:271: nll_loss_forward_reduce_cuda_kernel_2d: block: [0,0,0], thread: [0,0,0] Assertion `t >= 0 && t < n_classes` failed.
../aten/src/ATen/native/cuda/Loss.cu:271: nll_loss_forward_reduce_cuda_kernel_2d: block: [0,0,0], thread: [1,0,0] Assertion `t >= 0 && t < n_classes` failed.
../aten/src/ATen/native/cuda/Loss.cu:271: nll_loss_forward_reduce_cuda_kernel_2d: block: [0,0,0], thread: [2,0,0] Assertion `t >= 0 && t < n_classes` failed.
Epoch 1/9  ━━━━━━━━━━━━━━━━━╸━━━━━━━━━━━━━━━━━━━━━━ 57/127 0:02:20 • 0:02:53 0.40it/s loss: 1.88 v_num: yap0 val/loss: 2.658 val/acc: 0.316 val/acc_best: 0.316 train/loss: 2.005 train/acc: 0.229 
[2022-10-27 23:48:01,083][src.utils.utils][ERROR] -
```

This weirdly happens at the same point in the training iteration each time. I had been using this video reader for another training process just fine before.

I'm using pytorch-lightning BTW here's what my video reader pipeline looks like:

```
@pipeline_def
def video_pipe(file_list, stride, sequence_length, shuffle, num_shards, shard_id, reader_name):
    # stride 2 to skip every other frame. So 30fps becomes 15 fps
    video, label, start_frame_num, timestamps = fn.readers.video(device='gpu', file_list=file_list,
                                                                 sequence_length=sequence_length,
                                                                 shard_id=shard_id, num_shards=num_shards,
                                                                 initial_fill=10,
                                                                 image_type=types.RGB, dtype=types.FLOAT,
                                                                 file_list_frame_num=True,
                                                                 enable_frame_num=True,
                                                                 enable_timestamps=True,
                                                                 stride=stride,
                                                                 random_shuffle=shuffle,
                                                                 name=reader_name,
                                                                 pad_last_batch=True)

    label = fn.cast(label, dtype=types.INT64)

    return video, label, start_frame_num, timestamps


# here's what the pipeline
train_pipeline = video_pipe(batch_size=4, num_threads=8,
                                         device_id=0, shard_id=0,
                                         sequence_length=7,
                                         num_shards=0, stride=2, shuffle=True,
                                         file_list=str(data_path / "train_annotation.txt"), reader_name="train")

val_pipeline = video_pipe(batch_size=4, num_threads=8,
                                       device_id=0, shard_id=0,
                                       sequence_length=7,
                                       num_shards=1, stride=2, shuffle=False,
                                       file_list=str(data_path / "val_annotation.txt"), reader_name="val")

# test looks similar to val
```

Here's what the loader looks like:
```
train_loader = pytorch.DALIGenericIterator(pipelines=train_pipeline, auto_reset=True,
                                                    output_map=['videos', 'labels', 'start_frame_nums',
                                                                'timestamps'],
                                                    reader_name="train",
                                                    last_batch_policy=pytorch.LastBatchPolicy.DROP)


val_loader = pytorch.DALIGenericIterator(pipelines=val_pipeline, auto_reset=True,
                                                  output_map=['videos', 'labels', 'start_frame_nums', 'timestamps'],
                                                  reader_name="val",
                                                  last_batch_policy=pytorch.LastBatchPolicy.FILL)
```

And just for reference, this is what the annotation file follows this format:

```
../data/MERL_dataset_uncompressed/Videos_MERL_Shopping_Dataset/Videos_MERL_Shopping_Dataset/1_1_crop.mp4 0 0 155
../data/MERL_dataset_uncompressed/Videos_MERL_Shopping_Dataset/Videos_MERL_Shopping_Dataset/1_1_crop.mp4 0 212 216
../data/MERL_dataset_uncompressed/Videos_MERL_Shopping_Dataset/Videos_MERL_Shopping_Dataset/1_1_crop.mp4 0 372 375
../data/MERL_dataset_uncompressed/Videos_MERL_Shopping_Dataset/Videos_MERL_Shopping_Dataset/1_1_crop.mp4 1 417 417
../data/MERL_dataset_uncompressed/Videos_MERL_Shopping_Dataset/Videos_MERL_Shopping_Dataset/1_1_crop.mp4 1 460 526
../data/MERL_dataset_uncompressed/Videos_MERL_Shopping_Dataset/Videos_MERL_Shopping_Dataset/1_1_crop.mp4 2 562 565
../data/MERL_dataset_uncompressed/Videos_MERL_Shopping_Dataset/Videos_MERL_Shopping_Dataset/1_1_crop.mp4 2 715 738
../data/MERL_dataset_uncompressed/Videos_MERL_Shopping_Dataset/Videos_MERL_Shopping_Dataset/1_1_crop.mp4 3 765 783
../data/MERL_dataset_uncompressed/Videos_MERL_Shopping_Dataset/Videos_MERL_Shopping_Dataset/1_1_crop.mp4 4 895 913
../data/MERL_dataset_uncompressed/Videos_MERL_Shopping_Dataset/Videos_MERL_Shopping_Dataset/1_1_crop.mp4 5 940 940
../data/MERL_dataset_uncompressed/Videos_MERL_Shopping_Dataset/Videos_MERL_Shopping_Dataset/1_1_crop.mp4 5 980 1026

```

**Comment by JanuszL** [comment]
Hi @RafayAK,

Thank you for reporting this problem. Can you check if this problem is reproducible outside of the training (running the pipeline as a standalone entity)?
Can you provide [a minimal and complete reproduction script we can run on our side](https://github.com/NVIDIA/DALI#reporting-problems-asking-questions)?

**Comment by JanuszL** [comment]
Please reopen when you can provide more details regarding your issue.
